config.py

```diff
@@ -0,0 +1,51 @@
+import pydantic
+from typing import Optional, List, Dict, Union, Any
+from aws_cdk import aws_ec2
+
+
+class Config(pydantic.BaseSettings):
+    project_id: Optional[str] = pydantic.Field(
+        description="Project ID", default="cdk-eoapi-demo"
+    )
+    stage: Optional[str] = pydantic.Field(
+        description="Stage of deployment", default="test"
+    )
+    auth_provider_jwks_url: Optional[str] = pydantic.Field(
+        description="Auth Provider JSON Web Key Set URL for ingestion authentication. If not provided, no authentication will be required."
+    )
+    data_access_role_arn: Optional[str] = pydantic.Field(
+        description="Role ARN for data access, if none will be created at runtime.",
+    )
+    db_instance_type: Optional[str] = pydantic.Field(
+        description="Database instance type", default="t3.micro"
+    )
+    db_allocated_storage: Optional[int] = pydantic.Field(
+        description="Allocated storage for the database", default=5
+    )
+    public_db_subnet: Optional[bool] = pydantic.Field(
+        description="Whether to put the database in a public", default=False
+    )
+    nat_gateway_count: Optional[int] = pydantic.Field(
+        description="Number of NAT gateways to create", default=1
+    )
+    bastion_host_create_elastic_ip: Optional[bool] = pydantic.Field(
+        description="Whether to create an elastic IP for the bastion host",
+        default=False,
+    )
+    bastion_host_allow_ip_list: Optional[List[str]] = pydantic.Field(
+        description="YAML file containing list of IP addresses to allow SSH access to the bastion host",
+        default=[],
+    )
+    bastion_host_user_data: Optional[
+        Union[Dict[str, Any], aws_ec2.UserData]
+    ] = pydantic.Field(
+        description="Path to file containing user data for the bastion host",
+        default=aws_ec2.UserData.for_linux(),
+    )
+    titiler_buckets: Optional[List[str]] = pydantic.Field(
```

Comment:
@emileten @vincentsarago What do you guys think, I'd rather see us be consistent with the Ingestor here and use role injection and assumption to handle this.

Reply:
I agree that would make us more consistent. [That's how I set up the tiler construct though](https://github.com/developmentseed/cdk-pgstac/blob/c9aaddcf42cac6e1028ff733f99938885771c025/lib/titiler-pgstac-api/index.ts#L57), would need to make the change there as well.

Reply:
It actually would make sense to use the same role for both

Reply:
👍

Reply:
documented that in https://github.com/developmentseed/cdk-pgstac/issues/52